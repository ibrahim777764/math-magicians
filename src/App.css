.App {
  /* width: 700px; */
  width: 100%;
  display: flex;
  flex-direction: column;
}
