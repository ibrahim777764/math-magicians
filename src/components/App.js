/* eslint-disable no-unused-vars */
import React, { useState } from 'react';
import '../App.css';
import calculator from '../logic/calculate';
import ButtonPanel from './ButtonPanel';
import Display from './Display';

// class App extends Component {
//   constructor(props) {
//     super(props);
//     this.state = {
//       total: null,
//       next: null,
//       operation: null,
//     };
//   }

//   handleClick = buttonName => {
//     const calc = calculator.calculate(this.state, buttonName);
//     this.setState(calc);
//   }

//   render() {
//     const showResult = state => {
//       const { total, next, operation } = state;
//       const result = `${total}${operation}${next}`.replace(/null/g, '');
//       return result === '' ? undefined : result;
//     };
//     return (
//       <div className="App">

//         <Display result={showResult(this.state)} />
//         <ButtonPanel clickHandler={this.handleClick} />
//       </div>
//     );
//   }
// }

// export default App;

const App = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = buttonName => {
    const calc = calculator.calculate(state, buttonName);
    setState(calc);
  };

  const showResult = state => {
    const { total, next, operation } = state;
    const result = `${total}${operation}${next}`.replace(/null/g, '');
    return result === '' ? undefined : result;
  };
  return (
    <div>

      <Display result={showResult(state)} />
      <ButtonPanel clickHandler={handleClick} />

    </div>
  );
};

export default App;
