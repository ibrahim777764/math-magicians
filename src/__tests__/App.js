import '@testing-library/jest-dom';

import * as React from 'react';
import { render, screen } from '@testing-library/react';
import App from '../components/App';

describe('App', () => {
  test('renders App component', () => {
    const nonExist = 'No such text';

    render(<App />);

    expect(screen.queryByText(nonExist)).toBeNull();
  });
});
