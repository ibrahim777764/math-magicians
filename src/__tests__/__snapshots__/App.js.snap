// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
  data-testid="ancestor"
>
  <div
    className="App"
    data-testid="descendant"
  >
    <nav
      className="main-nav"
    >
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            id="logo"
            onClick={[Function]}
            style={Object {}}
          >
            Math Magicians
          </a>
        </li>
      </ul>
      <ul
        className="nav-links"
      >
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="content"
    >
      <main
        id="home"
      >
        <h1>
          Welcome to our page!
        </h1>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
        </p>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
        </p>
      </main>
    </div>
    <footer
      id="main-footer"
    >
      copyright
       
      ©
      2021
       
      |
       
      <span>
        Math Magician
      </span>
       
      all rights reserved.
    </footer>
  </div>
</div>
`;

exports[`App renders correctly 2`] = `
<div
  className="App"
  data-testid="ancestor"
>
  <div
    className="App"
    data-testid="descendant"
  >
    <nav
      className="main-nav"
    >
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            id="logo"
            onClick={[Function]}
            style={Object {}}
          >
            Math Magicians
          </a>
        </li>
      </ul>
      <ul
        className="nav-links"
      >
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="content"
    >
      <main
        id="home"
      >
        <h1>
          Welcome to our page!
        </h1>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
        </p>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
        </p>
      </main>
    </div>
    <footer
      id="main-footer"
    >
      copyright
       
      ©
      2021
       
      |
       
      <span>
        Math Magician
      </span>
       
      all rights reserved.
    </footer>
  </div>
</div>
`;

exports[`App renders correctly 3`] = `
<div
  className="App"
  data-testid="ancestor"
>
  <div
    className="App"
    data-testid="descendant"
  >
    <nav
      className="main-nav"
    >
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            id="logo"
            onClick={[Function]}
            style={Object {}}
          >
            Math Magicians
          </a>
        </li>
      </ul>
      <ul
        className="nav-links"
      >
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="content"
    >
      <main
        id="home"
      >
        <h1>
          Welcome to our page!
        </h1>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
        </p>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
        </p>
      </main>
    </div>
    <footer
      id="main-footer"
    >
      copyright
       
      ©
      2021
       
      |
       
      <span>
        Math Magician
      </span>
       
      all rights reserved.
    </footer>
  </div>
</div>
`;
